### ISSUE TITLE
Dynamic shapes in BEVFormer

### ISSUE THREAD

**Issue by meenakshiramanathan1** [tenstorrent/tt-forge-fe#1074]
## Description 

-  During the bringup of BEVFormer model (#937 ) faced dynamic shapes issue.
- Data mismatch was coming from a particular in-place assignment operation from spatial cross attention part

`for j in range(bs):
    for i, index_query_per_img in enumerate(indexes):
        slots[j, index_query_per_img] += queries[j, i, :len(index_query_per_img)]`

This operation was replaced using masked_scatter to eliminate the in-place assignement

`for j in range(bs):
    for i, index_query_per_img in enumerate(indexes):
                
        mask = torch.zeros_like(slots, dtype=torch.bool)
        mask[j, index_query_per_img] = True
        
        updates = slots[j, index_query_per_img] + queries[j, i, :len(index_query_per_img)]
        slots.masked_scatter_(mask, updates)`

But since a mask has been introduced to pick the indices to be updated dynamically, shape of indices during index_put op in tvm is (1, T.Any()) which is causing indexing issues during frontend conversion. This is turning out to be a blocker for this model.

## Reproduce

```
git checkout mramanathan/bevformer
pytest forge/test/models/pytorch/vision/bevformer/test_bevformer.py
```

**Comment by skhatriTT** [comment]
@meenakshiramanathan1 is this completed?
